### PR TITLE
Migrate TestAddonReviewCard.js

### LINKED ISSUE #11162: Migrate /components/TestAddonReviewCard.js to react-testing-library
Migrate /components/TestAddonReviewCard.js to react-testing-library

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #11162

### CODE REVIEW

File: tests/unit/amo/components/TestAddonReviewCard.js [hunk 1]
```diff
@@ -300,228 +363,215 @@ describe(__filename, () => {
...
-    );
+    render({ review });
 
-    const errorHandler = new ErrorHandler({
-      id: 'some-id',
-      dispatch: store.dispatch,
-    });
-    errorHandler.handle(new Error('some unexpected error'));
+    clickDeleteReview();
+    userEvent.click(screen.getByRole('button', { name: 'Delete' }));
```
Comment (eviljeff) [inline tests/unit/amo/components/TestAddonReviewCard.js:553]: where do we set up that the error occurs?  (The only reference to errors - `createFailedErrorHandler` - is after the click has taken place)

**Review by eviljeff — APPROVED** [review]
This was a long patch with a diff that didn't always match up so it's quite possible I overlooked something 😐

File: tests/unit/amo/components/TestAddonReviewCard.js [hunk 2]
```diff
@@ -300,228 +363,215 @@ describe(__filename, () => {
...
-    expect(dialog).toHaveProp('confirmButtonText', 'Delete review');
+    expect(
+      screen.getByRole('button', { name: 'Keep review' }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Delete review' }),
+    ).toBeInTheDocument();
   });
 
   it('does not render a confirmation message for slim=true', () => {
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply tests/unit/amo/components/TestAddonReviewCard.js:464]: It's just testing a different thing, but it's true that the setup is the same, so they could be combined into one test. Sometimes I have done that, and other times I did not. As you can see, this is a huge test file, and in fact I ended up writing a lot more code than this which will appear in future patches. So there were some times I took an opportunity to combine tests where it made sense, but other times when I figured it was best to just update tests to make them work and not bother to try to combine everything.

So the answer to your question is: they are testing different things.

**Comment by bobsilverberg** [comment]
> This was a long patch with a diff that didn't always match up so it's quite possible I overlooked something 😐

Yeah, I'm sorry this was such a big patch, but it's all one file, so it didn't make sense to do it in pieces. I understand it's difficult to review. I appreciate you taking a look at it. I have been very careful with this work, so hopefully nothing important was missed, and I am also checking the coverage reports to make sure I am not removing coverage when creating these patches.

**Pull request merged**